plutus-metatheory/src/Algorithmic/ReductionEC.lagda.md

```diff
@@ -305,52 +300,17 @@ data _—→⋆_ : {A : ∅ ⊢Nf⋆ *} → (∅ ⊢ A) → (∅ ⊢ A) → Set
     → (p : C ≡ _)
     → unwrap (wrap A B M) p —→⋆ substEq (∅ ⊢_) (sym p) M
 
-  β-sbuiltin : ∀{A B}
+  β-sbuiltin : ∀{A B}{tn}
       (b : Builtin)
     → (t : ∅ ⊢ A ⇒ B)
-    → ∀{az}
-    → (p : az <>> (Term ∷ []) ∈ arity b)
-    → (bt : BApp b p t) -- one left
+    → {pt : tn ∔ 0 ≣ fv♯ (signature b)} 
+    → ∀{an} → {pa : an ∔ 1 ≣  args♯ (signature b)}
+    → {σB : SigTy pt (bubble pa) B}
+    → (bt : BApp b (A B⇒ σB) t) -- one left
     → (u : ∅ ⊢ A)
     → (vu : Value u)
       -----------------------------
-    → t · u —→⋆ BUILTIN' b (bubble p) (BApp.step p bt vu)
-
-  β-sbuiltin⋆ : ∀{B : ∅ ,⋆ K ⊢Nf⋆ *}{C}
-      (b : Builtin)
-    → (t : ∅ ⊢ Π B)
-    → ∀{az}
-    → (p : az <>> (Type ∷ []) ∈ arity b)
-    → (bt : BApp b p t) -- one left
-    → ∀ A
-    → (q : C ≡ _)
-      -----------------------------
-    → t ·⋆ A / q —→⋆ BUILTIN' b (bubble p) (BApp.step⋆ p bt q)
-
-
-{-
--- does this need to be heterogeneous?
```

Comment:
there was something nasty happening here that I thought might be an agda bug and I never got to the bottom of.

Reply:
And now that is gone we will never know...